manifests/ca.pp

```diff
@@ -124,86 +131,158 @@
     mode   => '0750'
   })
 
-  exec { "copy easy-rsa to openvpn config folder ${name}":
-    command => "/bin/cp -r ${openvpn::params::easyrsa_source} ${etc_directory}/openvpn/${name}/easy-rsa",
-    creates => "${etc_directory}/openvpn/${name}/easy-rsa",
-    require => File["${etc_directory}/openvpn/${name}"],
-  }
-
-  file { [
-    "${etc_directory}/openvpn/${name}/easy-rsa/clean-all",
-    "${etc_directory}/openvpn/${name}/easy-rsa/build-dh",
-    "${etc_directory}/openvpn/${name}/easy-rsa/pkitool",
-  ]:
-    ensure  => file,
-    mode    => '0550',
-    require => Exec["copy easy-rsa to openvpn config folder ${name}"],
+  file { "${etc_directory}/openvpn/${name}/easy-rsa" :
+    ensure             => directory,
+    recurse            => true,
+    links              => 'follow',
+    source_permissions => 'use',
+    source             => "file:${openvpn::params::easyrsa_source}",
+    require            => File["${etc_directory}/openvpn/${name}"],
   }
 
   file { "${etc_directory}/openvpn/${name}/easy-rsa/revoked":
     ensure  => directory,
     mode    => '0750',
     recurse => true,
-    require => Exec["copy easy-rsa to openvpn config folder ${name}"],
+    require => File["${etc_directory}/openvpn/${name}/easy-rsa"],
   }
 
-  file { "${etc_directory}/openvpn/${name}/easy-rsa/vars":
-    ensure  => file,
-    mode    => '0550',
-    content => template('openvpn/vars.erb'),
-    require => Exec["copy easy-rsa to openvpn config folder ${name}"],
+  if $easyrsa_version == '2.0' {
```

Comment:
should we use[ versioncmp()](https://puppet.com/docs/puppet/5.5/function.html#versioncmp) here?

Reply:
I tryed to KISS. To use the function `versioncmp()` is some time usefull. But here, in my opinion it will not provide better result. But i'll do as you prefer. Tell me.